src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1496,7 +1496,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
 
                 case SymbolKind.Alias:
                     {
-                        var alias = symbol as AliasSymbol;
+                        var alias = (AliasSymbol)symbol;
                         symbol = alias.Target;
```

Comment:
> The perspective I've been thinking about this pattern is for Watson bugs.

I agree with that and that was one of the motivations behind the decision to disallow the pattern, perhaps I wasn't clear in expressing that in my previous message.